azurerm/resource_arm_cognitive_account_test.go

```diff
@@ -0,0 +1,115 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	// "github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// TODO: an update test going from External -> Internal -> External
+
+func TestAccAzureRMCognitiveAccount_basic(t *testing.T) {
+	log.SetOutput(os.Stdout)
+	resourceName := "azurerm_cognitive_account.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMCognitiveAccount_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAppCognitiveAccountDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCognitiveAccountExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAppCognitiveAccountDestroy(s *terraform.State) error {
+	// client := testAccProvider.Meta().(*ArmClient).cognitiveAccountsClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_cognitive_account" {
+			continue
+		}
+
+		// name := rs.Primary.Attributes["name"]
+		// resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		// ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		/*
+			resp, err := client.Get(ctx, resourceGroup, name)
+
+			if err != nil {
+				if utils.ResponseWasNotFound(resp.Response) {
+					return nil
+				}
+
+				return err
+			}
+		*/
+
+		return nil
+	}
+
+	return nil
+}
+
+func testCheckAzureRMCognitiveAccountExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		//rs, ok := s.RootModule().Resources[name]
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		/*appServiceEnvironmentName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for App Service Environment: %s", appServiceEnvironmentName)
+		}
+
+		client := testAccProvider.Meta().(*ArmClient).appServiceEnvironmentsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		resp, err := client.Get(ctx, resourceGroup, appServiceEnvironmentName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: App Service Environment %q (resource group: %q) does not exist", appServiceEnvironmentName, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on appServiceEnvironmentsClient: %+v", err)
+		}*/
```

Comment:
can we add this back in? This is used to ensure that the Cognitive Services account exists